src/io/fileops.c

```diff
@@ -246,6 +246,32 @@ MVMint64 MVM_file_isexecutable(MVMThreadContext *tc, MVMString *filename, MVMint
     }
 }
 #else
+
+static int are_we_group_member(gid_t group) {
+    int len;
+    gid_t *gids = 0;
+    int res;
+    int i;
+    // Check the user group.
+    if (getegid() == group) {
+        return 1;
+    }
+    // Check the supplementary groups.
+    do {
+        len  = getgroups(0, 0);
+        gids = MVM_realloc(gids, len * sizeof(gid_t));
+        res  = getgroups(len, gids);
+    } while (res < -1);
```

Comment:
Why the loop? The spec I found for `getgroups` says:

> If size is zero, list is not modified, but the total number of
> supplementary group IDs for the process is returned.  This allows the
> caller to determine the size of a dynamically allocated list to be
> used in a further call to getgroups().

Which would imply one call to get the length and another to populate the list would suffice. No loop = no risk of loop gone infinite by accident.

Reply:
The loop was my solution to the question of what to do with the return value. Should I just ignore the return value (according to doc the function can error when given an invalid pointer or `len` being to small) or somehow try to handle the error? How could a sane error handling look?

Reply:
> How could a sane error handling look?

Throw an exception (`MVM_exception_throw_adhoc`).

> the function can error when given an invalid pointer or len being to small

Given the pointer came straight out of `MVM_realloc`, then we can trust that. Given that we asked for the length above, then `len` being too small can only occur upon a race; I think we can just disregard such cases (e.g. ignore errors that are about the length, and throw on any others).

Also, if the first call returns `0`, then we can exit swiftly without the whole allocate step.

Reply:
Thanks for the detailed explanation! I changed it accordingly.